src/compiler/scanner.ts

```diff
@@ -1609,13 +1609,15 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
             case CharacterCodes.doubleQuote:
                 return '"';
             case CharacterCodes.u:
-                if (
-                    flags & EscapeSequenceScanningFlags.ScanExtendedUnicodeEscape &&
-                    pos < end && charCodeUnchecked(pos) === CharacterCodes.openBrace
-                ) {
+                if (pos < end && charCodeUnchecked(pos) === CharacterCodes.openBrace) {
                     // '\u{DDDDDD}'
                     pos -= 2;
-                    return scanExtendedUnicodeEscape(!!(flags & EscapeSequenceScanningFlags.ReportInvalidEscapeErrors));
+                    const result = scanExtendedUnicodeEscape(!!(flags & EscapeSequenceScanningFlags.ReportInvalidEscapeErrors));
+                    if (!(flags & EscapeSequenceScanningFlags.AllowExtendedUnicodeEscape)) {
+                        Debug.assert(flags & EscapeSequenceScanningFlags.ReportInvalidEscapeErrors);
+                        error(Diagnostics.Unicode_escape_sequences_are_only_available_when_the_Unicode_u_flag_or_the_Unicode_Sets_v_flag_is_set, start, pos - start);

```

Comment:
```suggestion
                        tokenFlags |= TokenFlags.ContainsInvalidEscape;
                        if (flags & EscapeSequenceScanningFlags.ReportInvalidEscapeErrors) {
                            error(Diagnostics.Unicode_escape_sequences_are_only_available_when_the_Unicode_u_flag_or_the_Unicode_Sets_v_flag_is_set, start, pos - start);
                        }
```

Previously we would have skipped over this, set `ContainsInvalidEscape`, and returned, so I don't think the `assert` is warranted.

Reply:
Well, currently in a regular expression, all errors are reported and all token flags set are ignored and reverted. Still, it seems worthwhile to apply your suggestion to prevent bugs in possible scenarios in the future.

Reply:
Well, currently in a regular expression, errors are always reported and all token flags set are ignored and reverted. Still, it seems worthwhile to apply your suggestion to prevent bugs in possible scenarios in the future.

Reply:
Odd, it looks like that caused the new tests to fail. I wonder if setting `ContainsInvalidEscape` is affecting whether we actually report the error.

Reply:
The new test errors look correct. It’s because you merged #58982 but the previous check was done before the merge.